spot_wrapper/calibration/calibration_util.py

```diff
@@ -0,0 +1,924 @@
+            # Compute the translations needed to transform to the center
+            x_trans = (charuco_board.getSquareLength() * charuco_board.getChessboardSize()[0]) / 2.0
+            y_trans = (charuco_board.getSquareLength() * charuco_board.getChessboardSize()[1]) / 2.0
+            # Adjust tvec to be relative to the center
+            center_trans = np.array([x_trans, y_trans, 0.0]).reshape((3, 1))
+            rmat, _ = cv2.Rodrigues(rvec)
+            tvec = tvec + rmat.dot(center_trans)
+
+            return np.array(rmat), np.array(tvec).ravel()
+        else:
+            raise ValueError(
+                "Corners were found, but failed to localize. You likely primed the robot too close to the board"
+            )
+    else:
+        raise ValueError(
+            "Couldn't detect any Charuco Boards in the image, "
+            "localization failed. Ensure the board is visible from the"
+            " primed pose."
+        )
+
+
+def convert_camera_t_viewpoint_to_origin_t_planning_frame(
+    origin_t_planning_frame: np.ndarray = np.eye(4),
+    planning_frame_t_opencv_camera: np.ndarray = np.eye(4),
+    opencv_camera_t_viewpoint: np.ndarray = np.eye(4),
+) -> np.ndarray:
+    """
+    Convert a camera viewpoint given in a camera frame, to the planning frame.
+
+    Camera frame should be in OpenCV/ ROS format, where
+    +x should point to the right in the image
+    +y should point down in the image
+    +z should point into to plane of the image.
+
+    Assuming that origin_t_planning_frame isn't aligned with this format, align it
+    so that the viewpoint can be localized as a hand pose command.
+
+    Args:
+        origin_t_planning_frame (np.ndarray, optional): 4x4 homogenous transform
+            from origin to planning frame. Defaults to np.eye(4).
+        planning_frame_t_opencv_camera (np.ndarray, optional): 4x4 homogenous transform
+            from planning frame to the camera. Defaults to np.eye(4).
+        opencv_camera_t_viewpoint (np.ndarray, optional): 4x4 homogenous transform
+            from camera to the future viewpoint. Defaults to np.eye(4).
+
+    Returns:
+        np.ndarray: 4x4 homogenous transform of origin to planning frame of future viewpoint.
+    """
+    origin_t_opencv_camera_pose = origin_t_planning_frame @ planning_frame_t_opencv_camera
+    origin_t_viewpoint = origin_t_opencv_camera_pose @ opencv_camera_t_viewpoint
+    origin_t_planning_frame_next = origin_t_viewpoint @ np.linalg.inv(planning_frame_t_opencv_camera)
+    return origin_t_planning_frame_next
+
+
+def get_relative_viewpoints_from_board_pose_and_param(
+    R_board: np.ndarray,
+    tvec: np.ndarray,
+    distances: np.ndarray = np.arange(0.5, 0.7, 0.1),
+    x_axis_rots: np.ndarray = np.arange(-20, 21, 5),
+    y_axis_rots: np.ndarray = np.arange(-20, 21, 5),
+    z_axis_rots: np.ndarray = np.array([-20, 21, 5]),
+    R_align_board_frame_with_camera: np.ndarray = np.array([[1, 0, 0], [0, -1, 0], [0, 0, -1]]),
```

Comment:
In general you don't want to use functions in default parameters. It would be better practice to have these be `Optional` with `None` as the default. Then below if None set to the values.

Reply:
An alternative would be to use a function to set variables in the module. That still runs the risk of your defaults getting modified as they are mutable